RecoMET/METProducers/src/METSignificanceProducer.cc

```diff
@@ -0,0 +1,108 @@
+// -*- C++ -*-
+//
+// Package:    METProducers
+// Class:      METSignificanceProducer
+//
+//
+
+//____________________________________________________________________________||
+#include "RecoMET/METProducers/interface/METSignificanceProducer.h"
+
+
+//____________________________________________________________________________||
+namespace cms
+{
+
+//____________________________________________________________________________||
+  METSignificanceProducer::METSignificanceProducer(const edm::ParameterSet& iConfig)
+  {
+    
+    std::vector<edm::InputTag> srcLeptonsTags = iConfig.getParameter< std::vector<edm::InputTag> >("srcLeptons");
+    for(std::vector<edm::InputTag>::const_iterator it=srcLeptonsTags.begin();it!=srcLeptonsTags.end();it++) {
+      lepTokens_.push_back( consumes<edm::View<reco::Candidate> >( *it ) );
+    }
+    
+    pfjetsToken_    = consumes<edm::View<reco::Jet> >(iConfig.getParameter<edm::InputTag>("srcPfJets"));
+
+    metToken_ = consumes<edm::View<reco::MET> >(iConfig.getParameter<edm::InputTag>("srcMet"));
+    pfCandidatesToken_ = consumes<edm::View<reco::Candidate> >(iConfig.getParameter<edm::InputTag>("srcPFCandidates"));
+
+    metSigAlgo_ = new metsig::METSignificance(iConfig);
+
+   produces<double>("METSignificance");
+   produces<double>("CovarianceMatrix00");
+   produces<double>("CovarianceMatrix01");
+   produces<double>("CovarianceMatrix10");
```

Comment:
uhm,
I thought these were supposed to be enough
https://github.com/cms-sw/cmssw/blob/CMSSW_7_4_X/DataFormats/Math/src/classes_def.xml#L11
```
 <class pattern="ROOT::Math::SMatrix<*>" />
 <class pattern="edm::Wrapper<*>" />
```

.. and we have ```  math::Error<2>::type e2; ``` in the classes.h already.
Your message has MatRepStd, which as I understand is a general 2x2.
Can you try math::Error<2>?